### ISSUE TITLE
Issue in joint.util.getByPath()

### ISSUE THREAD

**Issue by JoseGoncalves** [DavidDurman/joint#111]
I found a situation were I got the following error in joint.util.getByPath();

>Uncaught TypeError: Cannot use 'in' operator to search for 'input' in null

This can be solved by checking if `obj` is valid in the `keys` loop like this;

```javascript
            while (keys.length) {
                key = keys.shift();
                if (obj) {
                   if (key in obj) {
                        obj = obj[key];
                    } else {
                        return undefined;
                    }
                }
            }
```

**Comment by DavidDurman** [comment]
Do you have an example of the call?

**Comment by JoseGoncalves** [comment]
The situation was not directly caused by me, but from using the Inspector plugin with a list of objects configured as custom data. Adding two items by clicking in the plus button and then filling only the input(s) in the second item causes an array of 2 objects were the first one is set to `null`. After that If i click on my shape with this custom data attached, `joint.util.getByPath()` is called and crashes.

**Comment by DavidDurman** [comment]
Right. We'll have a look at this. We should fix the cause rather than patch `getByPath()` to cope with wrong parameters. The first parameter to `getByPath()` must be an object, not a falsy value. Thanks for reporting!

**Comment by kumilingus** [comment]
Hi Jose,
could you please provide us with the inspector input type definition?

**Comment by JoseGoncalves** [comment]
Sure;

```javascript
	var CommonInspectorInputs = {
		description: {
			type: 'textarea',
			label: 'Description',
			group: 'data',
			index: 1
		},
		descAsTip: {
			type: 'toggle',
			label: 'Description as Tooltip',
			group: 'data',
			index: 2
		},
		tooltip: {
			type: 'text',
			label: 'Tooltip',
			group: 'data',
			when: {
				eq: {
					'descAsTip': false
				}
			},
			index: 3
		},
	};

	var CommonInspectorGroups = {
		data: {
			label: 'Operator',
			index: 1
		}
	};

	joint.shapes.oper.InspectorDefs = {
		/* ... */
		'oper.Aggregation': {
			inputs: _.extend({
				attrs: {
					'.oper-label': {
						text: {
							type: 'text',
							label: 'Name',
							group: 'data',
							index: 4,
							attrs: {
								input: {
									pattern: '[_a-zA-Z][_a-zA-Z0-9]*',
									required: 'required'
								}
							},
						}
					}
				},
				config: {
					type: 'list',
					label: 'Aggregations',
					group: 'data',
					index: 5,
					item: {
						type: 'object',
						properties: {
							input: {
								type: 'text',
								label: 'Input',
								attrs: {
									input: {
										required: 'required'
									}
								},
							},
							operation: {
								type: 'select',
								label: 'Operation',
								options: ['GroupBy', 'Count', 'Sum', 'Avg', 'Max', 'Min'],
							},
							output: {
								type: 'text',
								label: 'Output',
								attrs: {
									input: {
										pattern: '[_a-zA-Z][_a-zA-Z0-9]*',
										required: 'required'
									}
								},
							}
						}
					}
				}
			}, CommonInspectorInputs),

			groups: CommonInspectorGroups
		}

	};
```

**Comment by kumilingus** [comment]
Thanks. I can't reproduce what you've described. The first item of the array built by Inspector is always an object.
````
{ input: '', operation: 'GroupBy', output: ''}
````
However I noticed that if I set `config` attribute manually to `[null]` this error occurs.
It's been fixed as you proposed with a tiny difference.
````
            while (keys.length) {
                key = keys.shift();
                if (Object(obj) === obj && key in obj) {
                    obj = obj[key];
                } else {
                    return undefined;
                }
             }
````
We don't want `joint.util.getByPath([null], '0.property')` to return `null` but `undefined` instead (similar to `joint.util.getByPath([], '0.property')`).
`joint.util.getByPath([5], '0.property')` also threw an error. That's why we are checking if `obj` is an object (`Object(a) ===  a`).

Cheers!
